packages/front-end/components/EventWebHooks/EventWebHookDetail/EventWebHookDetail.tsx

```diff
@@ -116,10 +167,58 @@ export const EventWebHookDetailContainer = () => {
   const router = useRouter();
   const { eventwebhookid: eventWebHookId } = router.query;
 
-  const { data, error } = useApi<{ eventWebHook: EventWebHookInterface }>(
-    `/event-webhooks/${eventWebHookId}`
+  const { apiCall } = useAuth();
+
+  const { data, error, mutate } = useApi<{
+    eventWebHook: EventWebHookInterface;
+  }>(`/event-webhooks/${eventWebHookId}`);
+
+  const [isEditModalOpen, setIsEditModalOpen] = useState<boolean>(false);
+  const [editError, setEditError] = useState<string | null>(null);
+
+  const handleEdit = useCallback(
+    async (data: EventWebHookEditParams) => {
+      if (!eventWebHookId) return;
+
+      // Keep the modal open and display error
+      const handleUpdateError = (message: string) => {
+        setEditError(`Failed to update webhook: ${message}`);
+        setIsEditModalOpen(true);
+      };
+
+      try {
+        const response = await apiCall<{ error?: string; status?: number }>(
+          `/event-webhooks/${eventWebHookId}`,
+          {
+            method: "PUT",
+            body: JSON.stringify(pick(data, ["events", "name", "url"])),
+          }
+        );
+
+        if (response.error) {
+          handleUpdateError(response.error || "Unknown error");
+        } else {
+          mutate();
+          setEditError(null);
+        }
+      } catch (e) {
+        handleUpdateError("Unknown error");
+      }
+    },
+    [mutate, apiCall, eventWebHookId]
   );
 
+  const handleDelete = useCallback(async () => {
+    if (!router) return;
+    if (!eventWebHookId) return;
+
+    await apiCall(`/event-webhooks/${eventWebHookId}`, {
+      method: "DELETE",
+    });
+
+    router.back();
```

Comment:
Nitpick but could we direct them to a specific page instead? This could lead to a weird edge case (in the case they opened a new tab, for example)

Reply:
Good call. I added a fix here 0a8d319228ae6c4dd1e38a8a784b5112e80398a6